packages/app/src/providers/h5grove/h5grove-api.ts

```diff
@@ -41,17 +48,22 @@ export class H5GroveApi extends ProviderApi {
   public async getValue(
     params: ValuesStoreParams
   ): Promise<H5GroveDataResponse> {
-    const { path } = params;
+    const { path, selection } = params;
     const entity = await this.getEntity(path);
     assertDataset(entity);
+    assertNonNullShape(entity);
 
     const DTypedArray = typedArrayFromDType(entity.type);
-    if (!DTypedArray) {
-      return this.fetchData(params);
+    if (DTypedArray) {
+      const buffer = await this.fetchBinaryData(params);
+      const array = new DTypedArray(buffer);
+      return hasScalarShape(entity) ? array[0] : [...array];
     }
 
-    const buffer = await this.fetchBinaryData(params);
-    return new DTypedArray(buffer);
+    const value = await this.fetchData(params);
+    return hasArrayShape(entity)
+      ? flattenValue(value, entity, selection)
+      : value;
```

Comment:
You could put the check inside `flattenValue`

Reply:
Hmm I think it's clearer if `flattenValue` accepts `Dataset<ArrayShape>` and returns `unknown[]` (which I'm going to make explicit), as it doesn't make sense to "flatten" a scalar.